src/main/java/io/dashboardhub/pipelinedashboard/config/UsernameAuditorAware.java

```diff
@@ -0,0 +1,19 @@
+package io.dashboardhub.pipelinedashboard.config;
+
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public class UsernameAuditorAware implements AuditorAware<String> {
+
+    public String getCurrentAuditor() {
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication == null || !authentication.isAuthenticated()) {
+            return null;
```

Comment:
Hmm, might be difficult here since we have no control over what the api returns. That said, this is a copy/paste from the tutorials on the net